File: haskell/tests/lehmer02.hs
import Groups
import Permutation (Permutation, showCycles, fromLehmer)

group :: Group Permutation
group = symmetric 5

main :: IO ()
main = mapM_ (\x -> let lehmer = gindex group x
			remhel = fromLehmer lehmer
		    in if x /= remhel then putStrLn $ showCycles x ++ " -> "
						   ++ show lehmer  ++ " -> "
						   ++ showCycles remhel
		       else return ()) $ gelems group

File: python2/tests/lehmer02.py
import sys
sys.path.insert(1, sys.path[0] + '/..')
from groups      import Symmetric
from permutation import Permutation

for x in Symmetric(5):
    lehmer = x.lehmer
    remhel = Permutation.fromLehmer(lehmer)
    if x != remhel:
	print '%s -> %d -> %s' % (x, lehmer, remhel)
